kymatio/scattering1d/core/scattering1d.py

```diff
@@ -91,26 +86,22 @@ def scattering1d(U_0, backend, filters, oversampling, average_local):
 
                 if j2 > j1:
                     # convolution + downsampling
-                    sub2_adj = min(j2, log2_T) if average_local else j2
-                    k2 = max(sub2_adj - k1 - oversampling, 0)
-
+                    sub2_adj = min(j2, log2_stride) if average_local else j2
+                    k2 = max(sub2_adj - k1, 0)
                     U_2_c = backend.cdgmm(U_1_hat, psi2[n2]['levels'][k1])
                     U_2_hat = backend.subsample_fourier(U_2_c, 2**k2)
-                    # take the modulus
                     U_2_c = backend.ifft(U_2_hat)
 
+                    # take the modulus
                     U_2_m = backend.modulus(U_2_c)
 
                     if average_local:
-                        U_2_hat = backend.rfft(U_2_m)
-
                         # Convolve with phi_J
-                        k2_log2_T = max(log2_T - k2 - k1 - oversampling, 0)
-
-                        S_2_c = backend.cdgmm(U_2_hat, phi['levels'][k1 + k2])
+                        U_2_hat = backend.rfft(U_2_m)
+                        k2_log2_T = max(log2_stride - sub2_adj, 0)
```

Comment:
Why no `-k1`?

Reply:
2**sub2_adj is the stride of U2 (granted, we could have a better name ...)
2**log2_stride is the stride of S2

going from U2 to S2 requires a subsampling of 2**(log2_stride - sub2_adj)